src/BrowserView.m

```diff
@@ -379,27 +382,32 @@ - (NSDragOperation)tabView:(NSTabView *)aTabView validateSlideOfProposedItem:(NS
 - (void)tabView:(NSTabView *)aTabView tabBarViewDidHide:(MMTabBarView *)tabBarView
 {
 	[self.tabBarHeightConstraint setConstant:0];
+	[APPCONTROLLER.mainWindowController showAddTabButtonInToolbar];
```

Comment:
I did not like this either. If I used `view.window.windowController` I would have to use a typecast. Same goes for the responder chain. Do you have a different suggestion on how to avoid this and the dependency?

Reply:
What do you think about using the NSNotificationCenter?

Reply:
The responder chain doesn’t require a typecast, just a selector.

You can use this method on BrowserView and then send a message using `@selector()`: https://developer.apple.com/documentation/appkit/nsapplication/1428509-sendaction

If BrowserView were in the view hierarchy, it would also work with https://developer.apple.com/documentation/appkit/nsresponder/1524516-trytoperform, as this explicitly sends a message to the window’s delegate.

Reply:
Ultimately we need a better model for Vienna entirely. I think we should use Storyboard and view controllers soon, as they make things a lot easier. But it’s still ver difficult to untangle the code.

Reply:
The problem is that this won´t solve the problem if we need a backwards reference to something like the window controller, as I know from developing for iOS. But I think the NSNotificationCenter is an appropriate solution for that.

Reply:
There are other solutions for that, such as delegates and dependency injection. NSNotificationCenter is too tempting and already (mis)used way too much in Vienna. At least for this PR, we don’t actually need a reference to the window controller.